crates/fuel-core/src/service/adapters/producer.rs

```diff
@@ -123,31 +123,53 @@ impl fuel_core_producer::ports::DryRunner for ExecutorAdapter {
 
 #[async_trait::async_trait]
 impl fuel_core_producer::ports::Relayer for MaybeRelayerAdapter {
-    async fn wait_for_at_least(
+    async fn get_latest_da_blocks_with_costs(
         &self,
-        height: &primitives::DaBlockHeight,
-    ) -> anyhow::Result<primitives::DaBlockHeight> {
+        starting_from: &DaBlockHeight,
+    ) -> anyhow::Result<Vec<(DaBlockHeight, u64)>> {
         #[cfg(feature = "relayer")]
         {
             if let Some(sync) = self.relayer_synced.as_ref() {
-                sync.await_at_least_synced(height).await?;
-                sync.get_finalized_da_height()
+                sync.await_at_least_synced(starting_from).await?;
+                let highest = sync.get_finalized_da_height()?;
+                (starting_from.0..=highest.0)
+                    .map(|height| get_gas_cost_for_height(height, sync))
```

Comment:
I'm afraid that the current usage of the Genesis da block height will lead to iterations from 0..5M.

It may not be a huge problem in the long term, but it may slow down the start of the `fuel core` for local development.